code/modules/antagonists/heretic/knowledge/knock_lore.dm

```diff
@@ -0,0 +1,230 @@
+		/datum/heretic_knowledge/codex_cicatrix,
+	)
+	cost = 1
+	route = PATH_KNOCK
+
+/datum/heretic_knowledge/knock_grasp/on_gain(mob/user, datum/antagonist/heretic/our_heretic)
+	RegisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK_SECONDARY, PROC_REF(on_secondary_mansus_grasp))
+	RegisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK, PROC_REF(on_mansus_grasp))
+
+/datum/heretic_knowledge/knock_grasp/on_lose(mob/user, datum/antagonist/heretic/our_heretic)
+	UnregisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK_SECONDARY)
+	UnregisterSignal(user, COMSIG_HERETIC_MANSUS_GRASP_ATTACK)
+
+/datum/heretic_knowledge/knock_grasp/proc/on_mansus_grasp(mob/living/source, mob/living/target)
+	SIGNAL_HANDLER
+	var/obj/item/clothing/under/suit = target.get_item_by_slot(ITEM_SLOT_ICLOTHING)
+	if(istype(suit) && suit.adjusted == NORMAL_STYLE)
+		suit.toggle_jumpsuit_adjust()
+		suit.update_appearance()
+
+/datum/heretic_knowledge/knock_grasp/proc/on_secondary_mansus_grasp(mob/living/source, atom/target)
+	SIGNAL_HANDLER
+
+	if(ismecha(target))
+		var/obj/vehicle/sealed/mecha/mecha = target
+		mecha.dna_lock = null
+		for(var/mob/living/occupant as anything in mecha.occupants)
+			if(isAI(occupant))
+				continue
+			mecha.mob_exit(occupant, randomstep = TRUE)
+	else if(istype(target,/obj/machinery/door/airlock))
+		var/obj/machinery/door/airlock/door = target
+		door.unbolt()
+	else if(istype(target, /obj/machinery/computer))
+		var/obj/machinery/computer/computer = target
+		computer.authenticated = TRUE
+		computer.balloon_alert(source, "unlocked")
+
+	var/turf/target_turf = get_turf(target)
+	SEND_SIGNAL(target_turf, COMSIG_ATOM_MAGICALLY_UNLOCKED, src, source)
+	playsound(target, 'sound/magic/hereticknock.ogg', 100, TRUE, -1)
+
+	return COMPONENT_USE_HAND
+
+/datum/heretic_knowledge/key_ring
+	name = "Key Keeper’s Burden"
+	desc = "Allows you to transmute a wallet, an iron rod, and an ID card to create an Eldritch Card. \
+		It functions the same as an ID Card, but attacking it with an ID card fuses it and gains its access. \
+		You can use it in-hand to change its form to a card you fused. \
+		Does not preserve the card used in the ritual."
+	gain_text = "Gateways shall open before me, my very will ensnaring reality."
+	adds_sidepath_points = 1
+	required_atoms = list(
+		/obj/item/storage/box = 1, //monkestation edit wallet ==> box (leather is too hard to get due to botany changes)
+		/obj/item/stack/rods = 1,
+		/obj/item/card/id = 1,
+	)
+	result_atoms = list(/obj/item/card/id/advanced/heretic)
+	next_knowledge = list(/datum/heretic_knowledge/limited_amount/rite_of_passage)
+	cost = 1
+	route = PATH_KNOCK
+
+/datum/heretic_knowledge/limited_amount/rite_of_passage // item that creates 3 max at a time heretic only barriers, probably should limit to 1 only, holy people can also pass
+	name = "Rite Of Passage"
+	desc = "Allows you to transmute a white crayon, a wooden plank, and a multitool to create a Consecrated Book. \
```

Comment:
change dis to a normal crayon in description, since you have changed its ingridient